datalake/src/main/java/com/sequenceiq/datalake/flow/upgrade/database/UpgradeDatabaseServerActions.java

```diff
@@ -0,0 +1,100 @@
+    public Action<?, ?> upgradeDatabase() {
+        return new AbstractUpgradeDatabaseServerSdxAction<>(SdxUpgradeDatabaseServerEvent.class) {
+
+            @Override
+            protected void doExecute(SdxContext context, SdxUpgradeDatabaseServerEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Execute upgrade database server flow for SDX: {}", payload.getResourceId());
+                sdxStatusService.setStatusForDatalakeAndNotify(DATALAKE_UPGRADE_DATABASE_SERVER_IN_PROGRESS, "Database server upgrade in progress",
+                        payload.getResourceId());
+                UpgradeDatabaseServerRequest request = UpgradeDatabaseServerRequest.from(context, payload.getTargetMajorVersion());
+                sendEvent(context, request);
+            }
+
+            @Override
+            protected Object getFailurePayload(SdxUpgradeDatabaseServerEvent payload, Optional<SdxContext> flowContext, Exception ex) {
+                return SdxUpgradeDatabaseServerFailedEvent.from(payload, ex, "Error launching database server upgrade");
+            }
+        };
+    }
+
+    @Bean(name = "SDX_UPGRADE_DATABASE_SERVER_FINISHED_STATE")
+    public Action<?, ?> finishedAction() {
+        return new AbstractUpgradeDatabaseServerSdxAction<>(SdxUpgradeDatabaseServerSuccessEvent.class) {
+
+            @Override
+            protected void doExecute(SdxContext context, SdxUpgradeDatabaseServerSuccessEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Database server upgrade finalized for SDX: {}", payload.getResourceId());
+                SdxCluster sdxCluster = sdxStatusService.setStatusForDatalakeAndNotify(DatalakeStatusEnum.RUNNING,
+                        "Database server upgrade completed successfully", payload.getResourceId());
+                metricService.incrementMetricCounter(MetricType.UPGRADE_DATABASE_SERVER_FINISHED, sdxCluster);
+                sendEvent(context, SDX_UPGRADE_DATABASE_SERVER_FINALIZED_EVENT.event(), payload);
+            }
+
+        };
+    }
+
+    @Bean(name = "SDX_UPGRADE_DATABASE_SERVER_FAILED_STATE")
+    public Action<?, ?> failedAction() {
+        return new AbstractUpgradeDatabaseServerSdxAction<>(SdxUpgradeDatabaseServerFailedEvent.class) {
+
+            @Override
+            protected void doExecute(SdxContext context, SdxUpgradeDatabaseServerFailedEvent payload, Map<Object, Object> variables) {
+                Exception exception = payload.getException();
+                DatalakeStatusEnum failedStatus = DatalakeStatusEnum.DATALAKE_UPGRADE_DATABASE_SERVER_FAILED;
+                LOGGER.info("Update SDX status to {} for resource: {}", failedStatus, payload.getResourceId(), exception);
+                String statusReason = "Database server upgrade failed";
+                if (exception.getMessage() != null) {
+                    statusReason = exception.getMessage();
+                }
```

Comment:
I think you shouldeither concat the 2 message or just use the tertiary operator when you initialize the variable